### ISSUE TITLE
HierarchicalCodebook.reproject_parents reassigns parents; breaks external references

### ISSUE THREAD

**Issue by sachncs** [sachncs/avqa#30]
## Problem

`src/avqa/codebook.py:211-217` `HierarchicalCodebook.reproject_parents` **reassigns** the parents attribute rather than modifying it in place:

```python
def reproject_parents(self) -> None:
    """Set each parent to the mean of its children (spec §7.9)."""
    self.parents = self.children.mean(dim=2)
```

Any external code holding a reference to the old `self.parents` tensor (e.g., a saved variable, a CUDA graph, an optimizer state) sees a stale tensor that no longer reflects the codebook's actual parents. The same pattern is used by `initialize_children_around_parents` (line 178) and `initialize_parents_random` (line 193).

## Evidence

- `src/avqa/codebook.py:211-217` (quoted above).
- `src/avqa/codebook.py:178-180` `initialize_children_around_parents` uses `self.parents.unsqueeze(2) + ...` which captures a reference to `self.parents` before `reproject_parents` runs.
- The optimizer state in `AVQAttention` (where `self.codebook.parents` is not an `nn.Parameter`, but `self.codebook.children` is referenced as a plain tensor) does not hold references across forward passes, but custom training loops that read `codebook.parents` after a BCAR step will see the wrong values.
- BCAR (`online_adaptation.py:127`) calls `parents.copy_(children.mean(dim=2))` directly — an in-place op. This is the safer pattern.

## Root Cause

The convenience method `reproject_parents` uses assignment rather than `copy_`, so external references go stale.

## Impact

- Code that does `old_parents = codebook.parents; reproject_parents(); use(old_parents)` is broken.
- State_dict round-trips are correct (the method detaches/clones), but custom checkpointing or live tensor references are not.

## Recommendation

Change `reproject_parents` to mutate `self.parents` in place:

```python
def reproject_parents(self) -> None:
    self.parents.copy_(self.children.mean(dim=2))
```

Same for `initialize_children_around_parents` (line 178): assign to `self.children.copy_(...)` rather than `self.children = ...`.

## Acceptance Criteria

- [ ] `reproject_parents` and `initialize_children_around_parents` use in-place updates.
- [ ] Add a test that asserts `old_ref is codebook.parents` after `reproject_parents()` (identity holds).
- [ ] `tests/unit/test_codebook.py` continues to pass; `validate_mean_constraint` still passes.

## Verification

```python
from avqa.codebook import HierarchicalCodebook
c = HierarchicalCodebook()
p = c.parents
c.reproject_parents()
print(p is c.parents)  # True after the fix
```

## Risk

Low. Behavioral equivalent; only the tensor identity changes.

**Comment by sachncs** [comment]
Resolved in PR #37.
